### ISSUE TITLE
perspective-view collapse / expand functions returning error

### ISSUE THREAD

**Issue by ccarli** [finos/perspective#1656]
## Bug Report
I'm trying to use the perspective-view [.collapse()](https://perspective.finos.org/docs/obj/perspective.html#module_perspective..view+collapse) method to displays the rows of my datagrid in a "grouped" state by default.
I'm not sure I'm using it correctly but when trying on one of the project's example I get the error message below

### Steps to Reproduce:
 1. open [Superstore example](https://bl.ocks.org/texodus/raw/803de90736a3641ad91c5c7a1b49d0a7/?raw=true)
 2. run in console 
```
    viewer = document.querySelector("perspective-viewer[name='Sales Report']")  
    table = await viewer.getTable()
    view = await table.view()
    await view.collapse()
```

### Expected Result:
update the datagrid display to group rows

### Actual Result:
no action on datagrid and error message below

![image](https://user-images.githubusercontent.com/30372857/145069461-50e23816-8d11-4b59-bf4e-c116e0b6b5a9.png)

same behaviour / error message with methods .expand() and .set_depth()

**Comment by texodus** [comment]
Thanks for the report, @ccarli!

There are a few things conflated here.  A `View` is kind of like a SQL query result set, but live - it holds the state and intermediate data for a set of query config values (e.g. what you put int he UI), including the expansions/collapse state of the tree (if it has one).  The line `view = await table.view()` in your example creates a _new_ `View`, e.g. not the one that your `<perspective-viewer>` UI created to render the data.  That causes two issues:

* This new `View` has no config options, it was instantiated without an argument in your example, `View()`.  Thus it is un-pivoted, and has no tree to collapse or expand, thus the error.  The error is unfortunately obtuse - we'll get that fixed.
* Even if it had been pivoted, `collapse()`/`expand()` are being called on the wrong `View`, and won't affect the `<perspective-viewer>` UI as written.

In addition, you'll need to call `.notifyResize(true)` subsequently on the `<perspective-viewer>` to force it to redraw itself, because expand/collapse do not trigger data notifications (currently).  So the correct code should look something like this:

```javascript
viewer = document.querySelector("perspective-viewer[name='Sales Report']");
view = await viewer.getView();
await view.collapse();
await viewer.notifyResize(true);
```

... However, it appears the `perspective-viewer` re-write released in `1.0.0+` did not implement this method `getView()`, so this is not working in the most recent release yet (In versions `<1.0.0`, the `.view` property exists which allows this).  I'll keep this report open to track this issue as well as the Error de-obfuscation issue above, these fixes should be in the `1.0.6` release.

**Comment by ccarli** [comment]
Very clear, thanks a lot!
